src/data_gradients/datasets/FolderProcessor.py

```diff
@@ -117,3 +116,107 @@ def __getitem__(self, index):
     def __iter__(self):
         for image_label_file in self.images_with_labels_files:
             yield image_label_file
+
+
+class ImageLabelConfigIterator:
+    """Iterate over all image and label files in the provided directories."""
+
+    def __init__(
+        self,
+        images_dir: str,
+        labels_dir: str,
+        config_path: str,
+        label_extensions: Sequence[str],
+        image_extensions: Sequence[str] = DEFAULT_IMG_EXTENSIONS,
+        verbose: bool = True,
+    ):
+        """
+        :param images_dir:      The directory containing the images.
+        :param labels_dir:      The directory containing the labels.
+        :param label_extensions: The extensions of the labels. Only the labels with this extensions will be considered.
+        :param image_extensions: The extensions of the images. Only the images with this extensions will be considered.
+        :param verbose:         Whether to print extra messages.
+        """
```

Comment:
Still need to add doc here. Basically, this Iterator works similarly to `ImageLabelFilesIterator` but only for file names listed in `config_path`

Reply:
Not sure I follow the value of both classes but this is a different subject...

Reply:
Are we sure we need to introduce a new class here ?

Reply:
I made it inherit from the other class here: https://github.com/Deci-AI/data-gradients/pull/117/commits/8fb32534cf09fa7288f7acaf3ce49dd07e22481e